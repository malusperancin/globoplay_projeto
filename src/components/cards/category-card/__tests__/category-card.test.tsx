/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";
import CategoryCardType from "../../category-card/category-card-type";
import CategoryCard from "../category-card";

describe("CategoryCard", () => {
  const card: CategoryCardType = {
    backgroundUrl: "teste",
    posterUrl: "teste",
    headline: "teste",
    id: "teste",
  };

  const categoryComponent = (
    <CategoryCard data={card} isFocus={true} translate={100} />
  );

  const categoryComponentNotFocus = (
    <CategoryCard data={card} isFocus={false} translate={100} />
  );

  it("should render without throwing an error", () => {
    const { container } = render(categoryComponent);

    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it("should be focus", () => {
    const { getByTestId } = render(categoryComponent);
    const focus =
      getByTestId("category-card").getElementsByClassName("buttonFocus")[0];

    expect(focus).toBeTruthy();
  });

  it("should not be focus", () => {
    const { getByTestId } = render(categoryComponentNotFocus);
    const focus =
      getByTestId("category-card").getElementsByClassName("button")[0];

    expect(focus).toBeTruthy();
  });
});
