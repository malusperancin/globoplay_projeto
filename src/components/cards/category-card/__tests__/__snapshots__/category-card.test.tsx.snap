// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard should render without throwing an error 1`] = `
<div>
  <div
    class="container"
    data-testid="category-card"
  >
    <button
      class="buttonFocus"
      style="transform: translate(100px);"
      type="button"
    >
      <img
        alt=""
        class="image"
        src="teste"
      />
    </button>
  </div>
</div>
`;
